// simple-login-form/dist/index.modern.js
import React, { Component } from 'react';

function _inheritsLoose(subClass, superClass) {
  subClass.prototype = Object.create(superClass.prototype);
  subClass.prototype.constructor = subClass;
  subClass.__proto__ = superClass;
}

var styles = {"profilephoto":"_rJjlp","formheading":"_3i9jh","formcontainer":"_3ctdA","inputsection":"_2_797","floatinglabel":"_aQJ47","checkboxcontainer":"_3mZ6m","loginbutton":"_1_cjN","loginformfooter":"_2Ip7F"};

var lock = require("./lock~GGVonpSK.png");

var user = require("./user~XDVrEXmo.png");

var profilePhoto = require("./seals~fVxqaeQu.jpg");

var SimpleLoginForm = /*#__PURE__*/function (_Component) {
  _inheritsLoose(SimpleLoginForm, _Component);

  function SimpleLoginForm() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _Component.call.apply(_Component, [this].concat(args)) || this;
    _this.state = {
      username: '',
      password: ''
    };

    _this.saveInfoFromInput = function (e) {
      if (e.target.name === 'username') {
        _this.setState({
          username: e.target.value
        });
      } else if (e.target.name === 'password') {
        _this.setState({
          password: e.target.value
        });
      }
    };

    _this.loadImage = function () {
      if (_this.props.photoIcon) {
        return _this.props.photoIcon;
      } else {
        return profilePhoto;
      }
    };

    return _this;
  }

  var _proto = SimpleLoginForm.prototype;

  _proto.render = function render() {
    var _this2 = this;

    return /*#__PURE__*/React.createElement("div", {
      style: this.props.style,
      className: styles.loginform
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.profilephoto
    }, /*#__PURE__*/React.createElement("img", {
      src: this.loadImage(),
      alt: ""
    })), /*#__PURE__*/React.createElement("div", {
      className: styles.formheading
    }, /*#__PURE__*/React.createElement("h1", null, "LOG IN")), /*#__PURE__*/React.createElement("div", {
      className: styles.formcontainer
    }, /*#__PURE__*/React.createElement("form", {
      onSubmit: function onSubmit(e) {
        return _this2.props.getUserInfo(_this2.state, e);
      }
    }, /*#__PURE__*/React.createElement("div", {
      className: styles.inputsection
    }, /*#__PURE__*/React.createElement("img", {
      src: user,
      alt: ""
    }), /*#__PURE__*/React.createElement("input", {
      type: "text",
      name: "username",
      onChange: function onChange(e) {
        return _this2.saveInfoFromInput(e);
      },
      required: true
    }), /*#__PURE__*/React.createElement("span", {
      className: styles.floatinglabel
    }, "Username")), /*#__PURE__*/React.createElement("div", {
      className: styles.inputsection
    }, /*#__PURE__*/React.createElement("img", {
      src: lock,
      alt: ""
    }), /*#__PURE__*/React.createElement("input", {
      type: "password",
      name: "password",
      onChange: function onChange(e) {
        return _this2.saveInfoFromInput(e);
      },
      required: true
    }), /*#__PURE__*/React.createElement("span", {
      className: styles.floatinglabel
    }, "Password")), /*#__PURE__*/React.createElement("div", {
      className: styles.loginbutton
    }, /*#__PURE__*/React.createElement("button", null, "LOGIN")), /*#__PURE__*/React.createElement("div", {
      className: styles.loginformfooter
    }, /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement("a", {
      href: "/sign-up"
    }, "Sign Up")), /*#__PURE__*/React.createElement("p", null, /*#__PURE__*/React.createElement("a", {
      href: "/forgot-password"
    }, "Forgot Password"))))));
  };

  return SimpleLoginForm;
}(Component);

export default SimpleLoginForm;
